### PR TITLE
feat(dbbrower): add method detailUsers in DBSchemaAccessor

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-feature, module-dbbrower
#### What this PR does / why we need it:
DBSchemaAccessor contains  method listUsers already,  but we may know detail information about user, eg user lock status 

#### Which issue(s) this PR fixes:

Add `detailUsers` method in DBSchemaAccessor to query user detail information.  current method return result is `List<DBUser>` which  add `DBUserLockType` to indicate user is locked or not.

#### Special notes for your reviewer:

OBMySQLSchemaAccessor  and  OBOracleSchemaAccessor implements DBSchemaAccessor  but MySQLSchemaAccessor and OracleSchemaAccessor don't,  because don't use this feature in mysql and oracle at present
 
#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/model/DBUserLockType.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+package com.oceanbase.tools.dbbrowser.model;
+
+/**
+ * @author yaobin
+ * @date 2023-09-27
+ * @since 4.2.3
+ */
+public enum DBUserLockType {
```
Comment (yizhouxw) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/model/DBUserLockType.java:24]: if locktype, then boolean type will by a simple model, e.g.  accountLocked.

however, the db-browser module was designed as an general model for RDBMS.
we may follow oracle design, `account_status` , refer https://docs.oracle.com/en/database/oracle/oracle-database/19/refrn/DBA_USERS.html#GUID-309FCCB2-2E8D-4371-9FC5-7F3B10E2A8C0 .

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/mysql/OBMySQLSchemaAccessor.java [hunk 2]
```diff
@@ -121,6 +123,23 @@ public List<DBObjectIdentity> listTables(String schemaName, String tableNameLike
         return results;
     }
 
+    @Override
+    public List<DBUser> detailUsers(List<String> usernames) {
+        MySQLSqlBuilder sb = new MySQLSqlBuilder();
+        sb.append("SELECT user_name, is_locked FROM oceanbase.__all_user");
```
Comment (yizhouxw) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/mysql/OBMySQLSchemaAccessor.java:129]: avoid internal table references. use compatible dictionary view instead.

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/mysql/OBMySQLSchemaAccessor.java [hunk 3]
```diff
@@ -121,6 +123,23 @@ public List<DBObjectIdentity> listTables(String schemaName, String tableNameLike
         return results;
     }
 
+    @Override
+    public List<DBUser> detailUsers(List<String> usernames) {
```
Comment (yizhouxw) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/mysql/OBMySQLSchemaAccessor.java:127]: query user detail by usernames list is not a common scenario.

what if listAllUsers ?

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/oracle/OBOracleSchemaAccessor.java [hunk 4]
```diff
@@ -137,6 +139,23 @@ public List<DBDatabase> listDatabases() {
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public List<DBUser> detailUsers(List<String> usernames) {
+        SqlBuilder sb = new OracleSqlBuilder();
+        sb.append("SELECT user_name, is_locked FROM SYS.ALL_VIRTUAL_USER_REAL_AGENT");
```
Comment (yizhouxw) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/oracle/OBOracleSchemaAccessor.java:145]: avoid internal view access

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/model/DBUserLockType.java (hunk 1 above)
Comment (krihy) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/model/DBUserLockType.java:24]: add DBUser in osc model which contains DBAccountLockType to indicate locked or not, and  DBAccountLockType can be easy transfer from db query result, and DBAccountLockType can convenient to show in log

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/mysql/OBMySQLSchemaAccessor.java (hunk 3 above)
Comment (krihy) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/mysql/OBMySQLSchemaAccessor.java:127]: this method `detailUsers` has placed in `OscDBAccessor` which is osc service code

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/oracle/OBOracleSchemaAccessor.java (hunk 4 above)
Comment (krihy) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/oracle/OBOracleSchemaAccessor.java:145]: see response above

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/mysql/OBMySQLSchemaAccessor.java (hunk 3 above)
Comment (krihy) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/mysql/OBMySQLSchemaAccessor.java:127]: This PR will be close, query user account lock status will put in osc service

**Pull request closed without merging**